system/include/GL/glfw.h

```diff
@@ -425,6 +425,27 @@ typedef void (GLFWCALL * GLFWthreadfun)(void *);
  * Prototypes
  *************************************************************************/
 
+#ifdef __EMSCRIPTEN__
+// Redirect GLFW2 symbols when they have different signatures under GLFW3.
+// This avoids the problem of JS symbols having different signatures in
+// different configurations.
+#define glfwGetWindowSize glfwGetWindowSize_v2
+#define glfwSetWindowSize glfwSetWindowSize_v2
+#define glfwSetWindowPos glfwSetWindowPos_v2
+#define glfwSetWindowTitle glfwSetWindowTitle_v2
+#define glfwIconifyWindow glfwIconifyWindow_v2
+#define glfwRestoreWindow glfwRestoreWindow_v2
+#define glfwSetWindowSizeCallback glfwSetWindowSizeCallback_v2
+#define glfwSetWindowCloseCallback glfwSetWindowCloseCallback_v2
+#define glfwSetWindowRefreshCallback glfwSetWindowRefreshCallback_v2
+#define glfwGetKey glfwGetKey_v2
+#define glfwGetMouseButton glfwGetMouseButton_v2
+#define glfwSetKeyCallback glfwSetKeyCallback_v2
+#define glfwSetCharCallback glfwSetCharCallback_v2
+#define glfwSetMouseButtonCallback glfwSetMouseButtonCallback_v2
+#define glfwSwapBuffers glfwSwapBuffers_v2
+#endif
```

Comment:
Interesting idea to have different names between 2/3. Makes sense to me. Do you think `_v2` is enough to differentiate them? I guess it is since nothing else should be using the `glfw` prefix at all?

Reply:
Yes,  I think its fine.  Not sure about poluting the public header like this, but it works.  If we wanted to do hide these symbol names we could write a `.c` file that maps the symbols to their v2 counterparts.